### PR TITLE
Fix broken unit tests + update CMake infra

### PR DESCRIPTION
**Pull request by vtangTT**
Included in PR: 
- CMake infra for tt-metal to include umd with `add_subdirectory`
- CMake infra for tests
- CPM for yaml-cpp and boost::interprocess
- Fix/re-enable GS/WH/BH unit tests

For BBE:
- I'm ensuring the make flow is not broken (running `make build` and `make test`)

### CODE REVIEW

File: tests/test_utils/generate_cluster_desc.hpp [hunk 1]
```diff
@@ -7,7 +7,39 @@
...
 #include <filesystem>
+#include <string>
+#include <iostream>
 
-namespace fs = std::filesystem;
+namespace test_utils {
 
-fs::path GetClusterDescYAML();
+inline std::string GetAbsPath(std::string path_){
+    std::filesystem::path umd_root = std::filesystem::absolute("umd" / std::filesystem::path(__FILE__).parent_path().parent_path().parent_path()).lexically_normal();
```
Comment (joelsmithTT) [inline tests/test_utils/generate_cluster_desc.hpp:16]: During the `make` build flow, this seems to result in a path e.g. `/home/joel/git/tt-umd/umd/device/bin/silicon/x86/create-ethernet-map` which is invalid.

**Review by joelsmithTT — APPROVED** [review]
Overall looks reasonable.  Some suggestions

- Augment readme with build instructions for CMake (command-line recipe for impatient people to copy/paste)
- Deprecate the Makefile build flow (it still needs to support BBE, up to you whether it correctly generates the unit tests)

**Comment by vtangTT** [comment]
> Overall looks reasonable. Some suggestions
> 
> * Augment readme with build instructions for CMake (command-line recipe for impatient people to copy/paste)
> * Deprecate the Makefile build flow (it still needs to support BBE, up to you whether it correctly generates the unit tests)

What do you mean by deprecate the makefile build flow? Remove it from the readme and replace it with cmake as the main build system?

File: tests/test_utils/generate_cluster_desc.hpp (hunk 1 above)
Comment (vtangTT) [reply tests/test_utils/generate_cluster_desc.hpp:16]: Will deprecate the make flow for unit tests if that's ok

**Pull request merged**